src/modeler/AutoRest.Swagger/Validation/SecurityDefinitionsStructureValidation.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
+using System;
+using System.Collections.Generic;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class SecurityDefinitionsStructureValidation : TypedRule<Dictionary<string, SecurityDefinition>>
+    {
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "M2054";
+
+        /// <summary>
+        /// Violation category of the Rule.
+        /// </summary>
+        public override ValidationCategory ValidationCategory => ValidationCategory.SDKViolation;
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Error;
+
+        /// <summary>
+        /// The template message for this Rule. 
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => Resources.SecurityDefinitionsStructureValidation;
+
+        /// <summary>
+        /// Checks for the presence and existence of the security definiton
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public override bool IsValid(Dictionary<string, SecurityDefinition> securityDefinitions)        
+        {
+            if (securityDefinitions.Count != 1)
+            {
+                return false;
+            }
+
+            foreach(KeyValuePair<string, SecurityDefinition> sd in securityDefinitions)
```

Comment:
You are really getting into LINQ.... :-) :-) Changed the code to use LINQ in all the places